lib/task_helpers/exports/provision_dialogs.rb

```diff
@@ -0,0 +1,27 @@
+module TaskHelpers
+  class Exports
+    class ProvisionDialogs
+      def export(options = {})
+        export_dir = options[:directory]
+
+        dialogs = if options[:all]
+                    MiqDialog.order(:id).all
+                  else
+                    MiqDialog.order(:id).where(:default => [false, nil])
+                  end
+
+        dialogs = dialogs.collect do |dialog|
```

Comment:
Could use collect! and not need the variable.

Reply:
`collect!` isn't defined for an `ActiveRecord::Relation []` so the only option is using `collect`

```
     NoMethodError:
       undefined method `collect!' for #<ActiveRecord::Relation []>
       Did you mean?  collect
```